lib/cli.js

```diff
@@ -68,7 +68,8 @@ function translateOptions(cliOptions) {
         allowInlineConfig: cliOptions.inlineConfig,
         reportUnusedDisableDirectives: cliOptions.reportUnusedDisableDirectives,
         resolvePluginsRelativeTo: cliOptions.resolvePluginsRelativeTo,
-        errorOnUnmatchedPattern: cliOptions.errorOnUnmatchedPattern
+        errorOnUnmatchedPattern: cliOptions.errorOnUnmatchedPattern,
+        stdin: cliOptions.stdin
```

Comment:
If it's okay to pass to pass this option to cli-engine we can solve the issue. However I am not sure, to solve a problem we pass this option that is required only for one specific use case and useless the rest of the times.

Reply:
This feels like an improvement, but I'm personally wary of adding more options to any of our public APIs just to handle one case like this.

One thought: Is there any way we can use the existing `CLIEngine` API to instantiate it with ignore patterns that negate the defaults?

If we are going to change the public API, I think we should add something more generic and self-explanatory like `noIgnore: boolean`.